connector_clickup/models/clickup_backend/common.py

```diff
@@ -0,0 +1,364 @@
+
+# Create a thread-local storage to hold the authorization code
+thread_local = threading.local()
+# Create a server-side storage to store the callback URL
+callback_storage = {}
+
+_logger = logging.getLogger(__name__)
+
+IMPORT_DELTA_BUFFER = 30  # seconds
+
+EXPORT_DELTA_BUFFER = 30  # seconds
+
+
+class ClickupBackend(models.Model):
+    _name = "clickup.backend"
+    _inherit = "connector.backend"
+    _description = "Clickup backend"
+
+    name = fields.Char()
+    auth_type = fields.Selection(
+        [("api_key", "API Key"), ("Oauth", "Oauth Authentication")],
+        required=True,
+        default="api_key",
+        help="Select Authentication Type",
+    )
+
+    datetime_filter_task = fields.Datetime(
+        help="""This will get all the tasks which is created
+         or updated after your provided date and time"""
+    )
+
+    force_update_tasks = fields.Boolean()
+    company_id = fields.Many2one(comodel_name="res.company", string="Company")
+
+    # Live
+    api_key = fields.Char(help="Enter API Key")
+    uri = fields.Char(help="Enter End Point Location Of Space Id")
+    client_id = fields.Char()
+    client_secret = fields.Char()
+    username = fields.Char()
+    password = fields.Char()
+    team_id = fields.Char()
+    # Test
+    test_mode = fields.Boolean(default=True)
+    test_token = fields.Char()
+    test_location = fields.Char()
+
+    def redirect_action(self):
+        html_file_path = "/connector_clickup/static/src/README.html"
+
+        return {
+            "type": "ir.actions.act_url",
+            "url": html_file_path,
+            "target": "new",
+        }
+
+    # def _compute_company_id(self):
+    #     for record in self:
+    #         record.company_id = self.env.company.id
+
+    # @api.onchange("company_id")
+    # def _onchange_company_id(self):
+    #     print("inside onchange method")
+    #     if self.company_id:
+    #         self.company_id.update({"clickup_backend_id": self.id})
+
+    def toggle_test_mode(self):
+        for record in self:
+            record.test_mode = not record.test_mode
+
+    def _import_from_date(
+        self,
+        model,
+        from_date_field=None,
+        filters=None,
+        force_update_field=None,
+        priority=None,
+        with_delay=True,
+    ):
+        """Common method for import data from from_date."""
+        if not filters:
+            filters = {}
+        import_start_time = datetime.now()
+        import_start_time_new = int(import_start_time.timestamp() * 1000)
+        job_options = {}
+        if priority or priority == 0:
+            job_options["priority"] = priority
+        for backend in self:
+            if from_date_field:
+                from_date = backend[from_date_field]
+                if from_date:
+                    from_date = int(from_date.timestamp() * 1000)
+
+                    filters = {"from_date": from_date, "to_date": import_start_time_new}
```

Comment:
use the update method of the dictionary to add values in filters, because it overrides the entire dictionary, and we lose track of old values.